### PR TITLE
Add CRUD endpoints for reefs

### PR DESCRIPTION
**Pull request by vickykont-programize**
This PR adds CRUD endpoints for reefs.

### CODE REVIEW

File: packages/api/src/reefs/reefs.service.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+      relations: ['region', 'admin', 'stream'],
+    });
+    if (!found) {
+      throw new NotFoundException(`Reef with ID ${id} not found.`);
+    }
+    return found;
+  }
+
+  async update(id: number, updateReefDto: UpdateReefDto) {
+    const result = await this.reefsRepository.update(id, updateReefDto);
```
Comment (bencpeters) [inline packages/api/src/reefs/reefs.service.ts:57]: Let's return the result - CRUD paths generally want to return the updated entity.

**Comment by bencpeters** [comment]
@vickykont-programize Let's just make sure the tests pass before merging - it looks like the Reefs controller needs a few more dependencies mocked before it gets injected.

File: packages/api/src/reefs/reefs.service.ts (hunk 1 above)
Comment (vickykont-programize) [reply packages/api/src/reefs/reefs.service.ts:57]: I looked into TypeORM's documentation and it provides 2 functions to handle updates, `save()` and `update()`. `save()` only returns some partial data for the updated entity and `update()` only some info about the query.

Should I add a second query to return the updated entity?

Comment (bencpeters) [reply packages/api/src/reefs/reefs.service.ts:57]: Ya, I think it probably makes the most sense to add a second query for the updated entity.

**Pull request merged**